src/profile_manager/forms.py

```diff
@@ -16,20 +18,62 @@ class Meta:
                   'visible_to_other_students', 'dark_theme', 'silent_mode', 'custom_stylesheet']
 
     def __init__(self, *args, **kwargs):
-        super(ProfileForm, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
+
         self.fields['grad_year'] = forms.ChoiceField(
             choices=Profile.get_grad_year_choices()
         )
+        
         self.fields['email'].initial = self.instance.user.email
 
     # UNIQUE if NOT NULL
     def clean_alias(self):
         return self.cleaned_data['alias'] or None
 
     def save(self, *args, **kwargs):
-        super(ProfileForm, self).save(*args, **kwargs)
+        super().save(*args, **kwargs)
+
         user = self.instance.user
         user.email = self.cleaned_data['email']
-        user.save()
+        user.save(update_fields=['email'])
+
+        return self.instance
+
+
+class UserForm(forms.ModelForm):
+    """ 
+        Staff only form for profile update view
+    """ 
+
+    is_TA = forms.BooleanField(required=False)
+
+    class Meta:
+        model = get_user_model()
+        fields = [
+            'username', 'first_name', 'last_name', 'is_staff', 'is_active', 'is_TA',
+        ]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        self.fields['username'].help_text = 'Unique usernames only!'
+
+        self.fields['is_TA'].label = 'TA'
+        self.fields['is_TA'].initial = self.instance.profile.is_TA
+        self.fields['is_TA'].help_text = Profile._meta.get_field('is_TA').help_text
+
+    def save(self):
+        # update self.instance since ProfileForm changes User model vars
+        # super().save() not saving email correctly so we only update the model fields in UserForm.fields
```

Comment:
added extra comment: ```# This code only updates the UserModel fields that is in self._meta.fields```

not really sure how else to explain the code

Reply:
Why is this required?  It's not normally required in a model form.  Something to do with when this form is combined with the UserForm in the same view and processed together right?  Can you explain that and the "Something"?

Reply:
ok ill add an extra comment: 
```# required since Userform saves over user.email field which is also a field saved in ProfileForm. This prevents email field from being overwritten in Userform```